src/containers/Tags.jsx

```diff
@@ -0,0 +1,254 @@
+  }
+
+  handleClose() {
+    this.setState({ openTagDialog: false });
+  }
+
+  async handleCreate(formData) {
+    await this.props.mutations.createTag(
+      formData,
+      this.props.params.organizationId
+    );
+    await this.props.data.refetch();
+    this.handleClose();
+  }
+
+  async handleEdit(formData) {
+    await this.props.mutations.editTag(
+      formData,
+      this.props.params.organizationId,
+      this.state.tagId
+    );
+    await this.props.data.refetch();
+    this.handleClose();
+  }
+
+  handleDelete(id) {
+    return async () => {
+      await this.props.mutations.deleteTag(
+        this.props.params.organizationId,
+        id
+      );
+      this.props.data.refetch();
+    };
+  }
+
+  render() {
+    const tags = this.props.data.tags.tags.sort((a, b) => a.id - b.id);
+    const { openTagDialog, dialogMode, tagId } = this.state;
+    const formSchema = yup.object({
+      name: yup.string().required(),
+      description: yup.string().nullable()
+    });
+    const dialogTag = tagId ? tags.find(t => t.id === tagId) : {};
+    return (
+      <div className={css(styles.cards)}>
+        {tags.map(t => (
+          <Card className={css(styles.card)} id={t.id} key={t.id}>
+            <CardText>
+              <Chip style={{ margin: 0 }} text={t.name} />
+            </CardText>
+            <CardText>{t.description}</CardText>
+            <CardText className={css(styles.buttons)}>
+              <RaisedButton
+                className={css(styles.editButton)}
+                primary
+                label="Edit"
+                labelPosition="before"
+                icon={<CreateIcon />}
+                onTouchTap={this.handleOpenEdit(t.id)}
+              />
+              <RaisedButton
+                label="Delete"
+                labelPosition="before"
+                icon={<DeleteIcon color={red500} />}
+                onTouchTap={this.handleDelete(t.id)}
+              />
+            </CardText>
+          </Card>
+        ))}
+        <FloatingActionButton
+          style={theme.components.floatingButton}
+          onTouchTap={this.handleOpen}
+        >
+          <ContentAdd />
+        </FloatingActionButton>
+        <Dialog
+          title={dialogMode === "create" ? "Create Tag" : "Edit Tag"}
+          open={openTagDialog}
+          onRequestClose={this.handleClose}
+        >
+          <GSForm
+            schema={formSchema}
+            onSubmit={
+              dialogMode === "create" ? this.handleCreate : this.handleEdit
+            }
+            defaultValue={dialogTag}
+          >
+            <div className={css(styles.fields)}>
+              <Form.Field label="Name" name="name" />
+              <Form.Field label="Description" name="description" />
+            </div>
+            <div className={css(styles.submit)}>
+              <Form.Button
+                type="submit"
+                label={dialogMode === "create" ? "Create" : "Save"}
```

Comment:
Would it make more sense to have `handleOpen` and `handleOpenEdit` set state values for `title`, `submitHandler`, and `buttonLabel`? It seems to me that would be cleaner (and more extensible) than the ternary expressions.

Reply:
added those, thanks